src/presenter/PresenterNodesOperationalSystem.cpp

```diff
@@ -162,13 +162,20 @@ PresenterNodesOperationalSystem::PresenterNodesOperationalSystem(
             Messages::OperationalSystemDownloadIso::Progress::download,
             selectedDistro->first, distroDownloadURL);
         m_view->progressMenu(Messages::title, desc.c_str(), std::move(command),
-            [](cloyster::CommandProxy& cmd) -> std::optional<double> {
+            [&](cloyster::CommandProxy& cmd) -> std::optional<double> {
                 auto out = cmd.getline();
                 if (!out) {
                     return std::nullopt;
                 }
                 std::string line = *out;
 
+                // If we have a line like ERROR 404: Not Found
+                // this means, obviously, that we did not found the URL.
+                if (line.contains("ERROR 404: Not Found")) {
+                    LOG_ERROR("URL {} not found", distroDownloadURL);
+                    return std::nullopt;
+                }
```

Comment:
I'm questioning if this is not the case for an exception? Is this nullopt handled by the UI?